cloud/services/scalesets/vmss_test.go

```diff
@@ -899,6 +899,185 @@ func TestReconcileVMSS(t *testing.T) {
 				}))
 			},
 		},
+		{
+			name:          "can create a vmss with encryption at host enabled",
+			expectedError: "",
+			expect: func(s *mock_scalesets.MockScaleSetScopeMockRecorder, m *mock_scalesets.MockClientMockRecorder) {
+				s.ScaleSetSpec().Return(azure.ScaleSetSpec{
+					Name:       "my-vmss",
+					Size:       "VM_SIZE_EAH",
+					Capacity:   2,
+					SSHKeyData: "ZmFrZXNzaGtleQo=",
+					OSDisk: infrav1.OSDisk{
+						OSType:     "Linux",
+						DiskSizeGB: 120,
+						ManagedDisk: infrav1.ManagedDisk{
+							StorageAccountType: "Premium_LRS",
+						},
+					},
+					SubnetName:                   "my-subnet",
+					VNetName:                     "my-vnet",
+					VNetResourceGroup:            "my-rg",
+					PublicLBName:                 "capz-lb",
+					PublicLBAddressPoolName:      "backendPool",
+					AcceleratedNetworking:        nil,
+					TerminateNotificationTimeout: to.IntPtr(7),
+					SecurityProfile:              &infrav1.SecurityProfile{EncryptionAtHost: to.BoolPtr(true)},
+				})
+				s.SubscriptionID().AnyTimes().Return("123")
+				s.ResourceGroup().AnyTimes().Return("my-rg")
+				s.V(gomock.AssignableToTypeOf(2)).AnyTimes().Return(klogr.New())
+				s.AdditionalTags()
+				s.Location().Return("test-location")
+				s.ClusterName().Return("my-cluster")
+				m.Get(context.TODO(), "my-rg", "my-vmss").
+					Return(compute.VirtualMachineScaleSet{}, autorest.NewErrorWithResponse("", "", &http.Response{StatusCode: 404}, "Not found"))
+				s.GetVMImage().Return(&infrav1.Image{
+					Marketplace: &infrav1.AzureMarketplaceImage{
+						Publisher: "fake-publisher",
+						Offer:     "my-offer",
+						SKU:       "sku-id",
+						Version:   "1.0",
+					},
+				}, nil)
+				s.GetBootstrapData(context.TODO()).Return("fake-bootstrap-data", nil)
+				m.CreateOrUpdate(context.TODO(), "my-rg", "my-vmss", gomockinternal.DiffEq(compute.VirtualMachineScaleSet{
```

Comment:
instead of checking the entire struct here since we only care about testing the encryption at host functionality is enabled it would make sense to do the same thing you did in the virtualmachines test https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1012/files#diff-f2a0391546f379219695118e52694fe6a8c5739619cb672941617a78d0c63336R770

Reply:
Updated the test... maybe not in the best possible way, though. I had to modify the test case interface a bit to pass through the `*WithT` pointer. Happy to change it if needed.